### PR TITLE
raftstore: use channel instead of vecque

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang PTAL

### CODE REVIEW

File: src/raftstore/store/worker/mod.rs [hunk 1]
```diff
@@ -1,86 +1,109 @@
...
-            stopped: Arc::new(RwLock::new(false)),
         }
     }
 
-    pub fn start<R: Runnable<T> + Send + 'static>(&mut self, mut runner: R) -> io::Result<()> {
+    pub fn start<R: Runnable<T> + Send + 'static>(&mut self, mut runner: R) -> Result<()> {
         info!("starting working thread: {}", self.name);
-        let stopped = self.stopped.clone();
-        let queue = self.queue.clone();
+        if self.receiver.is_none() {
```
Comment (ngaut) [inline src/raftstore/store/worker/mod.rs]: is_some ?

File: src/raftstore/store/worker/mod.rs [hunk 2]
```diff
@@ -1,86 +1,109 @@
...
 use std::io;
 use std::fmt::Display;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::mpsc::{self, Sender, Receiver};
 use std::time::Instant;
-
-use util::HandyRwLock;
+use std::result;
 
 const TASK_PEEK_INTERVAL_SECS: u64 = 100;
```
Comment (ngaut) [inline src/raftstore/store/worker/mod.rs]: Remove unused constant.

File: src/raftstore/store/worker/mod.rs (hunk 1 above)
Comment (BusyJay) [inline src/raftstore/store/worker/mod.rs]: No, here we check whether the background thread started. `receiver` will be none if it has started.

**Comment by ngaut** [comment]
LGTM

**Comment by siddontang** [comment]
I think now we can move work to util so that other like Peer Resolver can use it too.

**Pull request merged**